test/mainnet/ynclisbnb.spec.sol

```diff
@@ -0,0 +1,583 @@
+        );
+        assertEq(totalSupplyAfter, totalSupplyBefore, "Total supply of vault should not change");
+        assertEq(slisBnbLockedInVaultAfter, slisBnbLockedInVaultBefore, "SlisBnb locked in vault should not change");
+        assertEq(
+            clisBnb.balanceOf(MC.YIELDNEST_MPC_WALLET),
+            clisBnbBalanceOfYieldnestMpcWalletBefore,
+            "ClisBnb balance of YieldnestMpcWallet should not change"
+        );
+    }
+
+    function test_Vault_Reward_Stream_Allocation(uint256 depositAmount, uint256 rewardAmount) public {
+        rewardAmount = bound(rewardAmount, 0.1 ether, 1000000 ether);
+
+        test_Vault_Reward_Stream(depositAmount, rewardAmount);
+
+        uint256 totalAssets = clisBnbStrategy.totalAssets();
+        uint256 totalSupply = clisBnbStrategy.totalSupply();
+        uint256 clisBnbStrategyRate = clisBnbStrategy.previewRedeem(1 ether);
+        uint256 slisBnbLockedInVaultBefore =
+            _getStakedSlisBnbBalanceByVault(address(baseAsset), address(clisBnbStrategy));
+        uint256 clisBnbBalanceOfYieldnestMpcWalletBefore = clisBnb.balanceOf(MC.YIELDNEST_MPC_WALLET);
+
+        {
+            vm.startPrank(PROCESSOR);
+            address[] memory targets = new address[](2);
+            uint256[] memory values = new uint256[](2);
+            bytes[] memory datas = new bytes[](2);
+            targets[0] = address(slisBnb);
+            values[0] = 0;
+            datas[0] = abi.encodeWithSelector(IERC20.approve.selector, MC.SLIS_BNB_PROVIDER, rewardAmount);
+            targets[1] = MC.SLIS_BNB_PROVIDER;
+            values[1] = 0;
+            datas[1] = abi.encodeWithSelector(ISlisBnbProvider.provide.selector, rewardAmount, MC.YIELDNEST_MPC_WALLET);
+            clisBnbStrategy.processor(targets, values, datas);
+            vm.stopPrank();
+        }
+
+        uint256 clisBnbStrategyRateAfter = clisBnbStrategy.previewRedeem(1 ether);
+        uint256 totalAssetsAfter = clisBnbStrategy.totalAssets();
+        uint256 totalSupplyAfter = clisBnbStrategy.totalSupply();
+        uint256 slisBnbLockedInVaultAfter =
+            _getStakedSlisBnbBalanceByVault(address(baseAsset), address(clisBnbStrategy));
+        assertEq(clisBnbStrategyRateAfter, clisBnbStrategyRate, "ClisBnb strategy rate should not change");
+        assertEq(totalAssetsAfter, totalAssets, "Total assets of vault should not change");
+        assertEq(totalSupplyAfter, totalSupply, "Total supply of vault should not change");
+        assertEq(
+            slisBnbLockedInVaultAfter,
+            slisBnbLockedInVaultBefore + rewardAmount,
+            "SlisBnb locked in vault should increase by reward amount"
+        );
+        assertApproxEqRel(
+            clisBnb.balanceOf(MC.YIELDNEST_MPC_WALLET),
+            clisBnbBalanceOfYieldnestMpcWalletBefore + rewardAmount,
+            5e16,
+            "ClisBnb balance of YieldnestMpcWallet should increase by 95% of reward amount"
+        );
+    }
+
```

Comment:
can we add a test that withdraws after `test_Vault_Reward_Stream_Allocation`

so it withdraws with rewards and asserts things are ok